cursorless.nvim/lua/cursorless/init.lua

```diff
@@ -0,0 +1,131 @@
+-- local Config = require('cursorless.config')
+
+-- We can't use that yet as then when we use load_extensions() we get an error.
+-- So instead atm we rely on registering the functions from vim script
+-- local function register_functions1()
+--   local path = require('cursorless.utils').cursorless_nvim_path()
+--   vim.api.nvim_call_function('remote#host#RegisterPlugin', {
+--     'node',
+--     path .. '/rplugin/node/command-server',
+--     {
+--       { sync = false, name = 'CommandServerLoadExtension', type = 'function', opts = {} },
+--       { sync = false, name = 'CommandServerRunCommand', type = 'function', opts = {} },
+--     },
+--   })
+--   vim.api.nvim_call_function('remote#host#RegisterPlugin', {
+--     'node',
+--     path .. '/rplugin/node/cursorless-neovim',
+--     {
+--       { sync = false, name = 'CursorlessLoadExtension', type = 'function', opts = {} },
+--     },
+--   })
+--   vim.api.nvim_call_function('remote#host#RegisterPlugin', {
+--     'node',
+--     path .. '/rplugin/node/neovim-registry',
+--     {},
+--   })
+-- end
+
+local function register_functions()
+  local path = require('cursorless.utils').cursorless_nvim_path()
+  -- revert to using forward slashes as works when passed to remote#host#RegisterPlugin()
+  if require('cursorless.utils').is_win() then
+    path = path:gsub('\\', '/')
+  end
+  vim.api.nvim_call_function('RegisterFunctions', { path })
+end
+
+-- this triggers loading the node process as well as calling one function
+-- in the cursorless-neovim, command-server and neovim-registry extensions
+-- in order to initialize them
+local function load_extensions()
+  vim.api.nvim_call_function('CursorlessLoadExtension', {})
+  vim.api.nvim_call_function('CommandServerLoadExtension', {})
```

Comment:
so the idea is to just ship command server with cursorless for now and load it here?